### ISSUE TITLE
No way to write async generators

### ISSUE THREAD

**Issue by jimmywarting** [sindresorhus/eslint-plugin-unicorn#1196]
```js
class Foo {
  * bar() {
  }
}
```
  ✖  96:10  Unexpected space before function parentheses.  space-before-function-paren

```js
class Foo {
  *bar() {
  }
}
```
✖  96:8   Missing space after *

How da --- do you want me to write generators? the linter complain on both formats
same goes for using `async * foo() { ... }`
<details><summary>rant!</summary>I literally hate linters cuz they enforce code formatting, they change all the time and so dose your preferred style of coding over the years
i.m.o linters should <strong>only detect errors<strong> such as

- Variable already defined
- duplicate args or keys in objects
- warn about depreciations
- required full/relative path with extension for ESM to make it browser compatible
- and any kind of **errors**

if i want to use `==`, `var`, skip `;` or write any valid syntax - then i want to do so! 
With a 10+ years of javascript experiences then you know what you are doing and why you are doing it. 

The only thing i want a linter to do is to detect errors. Not predict them or enforce how i format my code for consistency. Isn't there any such rule list somewhere?

Sorry but i get mad sometimes when I'm put into a project that are not my own, you end up spending a heck a lot of time trying to make the linter happy and makes you frustrated cuz one person defined that ruleset and then that person quits his job so you lose your concentration
</details>

**Comment by sindresorhus** [comment]
How is this related to `eslint-plugin-unicorn`?

**Comment by fisker** [comment]
Close this since it's not related to this repo.
